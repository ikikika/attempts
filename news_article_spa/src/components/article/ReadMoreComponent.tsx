import React from "react";
import { ChevronRight } from "react-bootstrap-icons";

const styles = {
  wrapper: {
    backgroundColor: "var(--c-muted)",
  },
  textNormal: {
    fontSize: "var(--fs-article-cta)",
    fontWeight: 500,
    fontFamily: "var(--ff-article-cta)",
    lineHeight: "var(--lh-article-cta)",
  },
  textEmphasis: {
    fontSize: "var(--fs-category-l)",
    fontWeight: 600,
    fontFamily: "var(--ff-category-l)",
    lineHeight: "var(--lh-category-l)",
    textTransform: "uppercase" as "uppercase",
    color: "var(--c-primary)",
  },
  textCta: {
    fontSize: "var(--fs-button-s)",
    fontWeight: 500,
    fontFamily: "var(--ff-button-s)",
    lineHeight: "var(--lh-button-s)",
    color: "var(--c-primary)",
  },
};

const ReadMoreComponent = () => {
  return (
    <a
      href="https://www.todayonline.com/taxonomy/term/14366"
      style={styles.wrapper}
      className="px-4 py-3 mb-4 d-flex flex-row justify-content-between align-items-center"
    >
      <div className="d-flex flex-column flex-md-row align-items-center">
        <span className="me-1" style={styles.textNormal}>Read more of the latest in</span>
        <span style={styles.textEmphasis}>Singapore</span>
      </div>
      <div style={styles.textCta}>
        Explore now <ChevronRight />
      </div>
    </a>
  );
};

export default ReadMoreComponent;
