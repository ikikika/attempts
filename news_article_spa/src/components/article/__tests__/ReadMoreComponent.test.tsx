import React from "react";
import renderer from "react-test-renderer";

import ReadMoreComponent from "../ReadMoreComponent";

it("ReadMoreComponent matches snapshot", () => {
  const tree = renderer.create(<ReadMoreComponent />).toJSON();
  expect(tree).toMatchSnapshot();
});
