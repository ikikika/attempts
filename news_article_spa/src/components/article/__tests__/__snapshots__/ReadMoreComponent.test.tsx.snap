// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReadMoreComponent matches snapshot 1`] = `
<a
  className="px-4 py-3 mb-4 d-flex flex-row justify-content-between align-items-center"
  href="https://www.todayonline.com/taxonomy/term/14366"
  style={
    Object {
      "backgroundColor": "var(--c-muted)",
    }
  }
>
  <div
    className="d-flex flex-column flex-md-row align-items-center"
  >
    <span
      className="me-1"
      style={
        Object {
          "fontFamily": "var(--ff-article-cta)",
          "fontSize": "var(--fs-article-cta)",
          "fontWeight": 500,
          "lineHeight": "var(--lh-article-cta)",
        }
      }
    >
      Read more of the latest in
    </span>
    <span
      style={
        Object {
          "color": "var(--c-primary)",
          "fontFamily": "var(--ff-category-l)",
          "fontSize": "var(--fs-category-l)",
          "fontWeight": 600,
          "lineHeight": "var(--lh-category-l)",
          "textTransform": "uppercase",
        }
      }
    >
      Singapore
    </span>
  </div>
  <div
    style={
      Object {
        "color": "var(--c-primary)",
        "fontFamily": "var(--ff-button-s)",
        "fontSize": "var(--fs-button-s)",
        "fontWeight": 500,
        "lineHeight": "var(--lh-button-s)",
      }
    }
  >
    Explore now 
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 16 16"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M4.646 1.646a.5.5 0 0 1 .708 0l6 6a.5.5 0 0 1 0 .708l-6 6a.5.5 0 0 1-.708-.708L10.293 8 4.646 2.354a.5.5 0 0 1 0-.708z"
        fillRule="evenodd"
      />
    </svg>
  </div>
</a>
`;
